app/scripts/lib/rpc-method-middleware/handlers/wallet-createSession/handler.ts

```diff
@@ -117,55 +143,150 @@ async function walletCreateSessionHandler(
       }
     };
 
+    // if solana is a requested scope but not supported, we add a promptToCreateSolanaAccount flag to request
+    const isSolanaRequested =
+      Object.keys(requiredScopesWithSupportedMethodsAndNotifications).some(
+        (key) => {
+          let namespace = '';
+          try {
+            namespace = parseCaipChainId(key as CaipChainId).namespace;
+          } catch (err) {
+            return false;
+          }
+          return namespace === KnownCaipNamespace.Solana;
+        },
+      ) ||
+      Object.keys(optionalScopesWithSupportedMethodsAndNotifications).some(
+        (key) => {
+          let namespace = '';
+          try {
+            namespace = parseCaipChainId(key as CaipChainId).namespace;
+          } catch (err) {
+            return false;
+          }
+          return namespace === KnownCaipNamespace.Solana;
+        },
+      );
```

Comment:
let's make a helper that checks if a namespace exists in ScopesObject

Reply:
Working on this one now

Reply:
PR here: https://github.com/MetaMask/metamask-extension/pull/31608

Reply:
merged